modules/bioinspired/src/transientareassegmentationmodule.cpp

```diff
@@ -454,7 +454,7 @@ void TransientAreasSegmentationModuleImpl::_run(const std::valarray<float> &inpu
 
     // first square the input in order to increase the signal to noise ratio
     // get motion local energy
-    _squaringSpatiotemporalLPfilter(&inputToSegment[channelIndex*getNBpixels()], &_localMotion[0]);
+    _squaringSpatiotemporalLPfilter(&const_cast<std::valarray<float>&>(inputToSegment)[channelIndex*getNBpixels()], &_localMotion[0]);
```

Comment:
@mshabunin Do you want me to make this change or does everything else look good? Cheers for checking everything so thoroughly by the way!